### PR TITLE
Use ADAL 3 and Microsoft.Rest.ClientRuntime.Azure.Authentication

### PR DESCRIPTION
**Pull request by haocs**

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@haocs__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <p>
        It looks like you're working at Microsoft (hach). If you're full-time, we DON'T require a contribution license agreement.
    </p>
    <p>
        If you are a vendor, DO please sign the electronic contribution license agreement. It will take 2 minutes and there's no faxing! https://cla.azure.com.
    </p>

TTYL, AZPRBOT;

**Comment by hovsepm** [comment]
@azuresdkci add to whitelist

File: src/Common/Authentication/Authentication.sln [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+# Visual Studio 2013
+VisualStudioVersion = 12.0.40629.0
+MinimumVisualStudioVersion = 10.0.40219.1
+Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "Common.Authentication", "Common.Authentication\Common.Authentication.csproj", "{B95C489C-8CB7-4DCF-8D5F-B9AEBDBBAF89}"
+EndProject
+Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "Common.Authentication.Tests", "Common.Authentication.Tests\Common.Authentication.Tests.csproj", "{C2CF99A2-D35E-4AED-AFB9-C26960AF1D0D}"
+EndProject
+Global
+	GlobalSection(SolutionConfigurationPlatforms) = preSolution
+		Net40-Debug|Any CPU = Net40-Debug|Any CPU
```
Comment (stankovski) [inline src/Common/Authentication/Authentication.sln]: There is no need for Net40 and Net45 build configurations (we should eventually remove them from AutoRest). In PowerShell it's just Debug and Release.

File: src/Common/Authentication/Common.Authentication/Utilities/FileUtilities.cs [hunk 2]
```diff
@@ -0,0 +1,321 @@
...
+using System;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using System.Text;
+
+namespace Microsoft.Azure.Common.Authentication
+{
+    public static class FileUtilities
```
Comment (stankovski) [inline src/Common/Authentication/Common.Authentication/Utilities/FileUtilities.cs]: These do not sound like they belong to Authentication. Would be better if these extensions lived in Common.

File: src/Common/Authentication/Common.Authentication/Utilities/DictionaryExtensions.cs [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Microsoft.Azure.Common.Authentication.Utilities
+{
+    public static class DictionaryExtensions
```
Comment (stankovski) [inline src/Common/Authentication/Common.Authentication/Utilities/DictionaryExtensions.cs]: These do not sound like they belong to Authentication. Would be better if these extensions lived in Common.

**Pull request merged**